llvm/lib/CodeGen/CMakeLists.txt

```diff
@@ -50,6 +50,7 @@ add_llvm_component_library(LLVMCodeGen
   DeadMachineInstructionElim.cpp
   DetectDeadLanes.cpp
   DFAPacketizer.cpp
+  DroppedVariableStats.cpp
```

Comment:
I think the linking error occurs because this pass is in a CodeGen library that IR libraries do not (and should not) depend on.
Building with `-DBUILD_SHARED_LIBS=ON` passed to CMake should reproduce the issue.